src/state.rs

```diff
@@ -0,0 +1,22 @@
+/// The state of a single unit
+#[derive(RustcEncodable, RustcDecodable, PartialEq)]
+pub struct Unit {
+    pub position: [u64; 2],
+    pub angle: u64,
+    pub speed_vector: [u64; 2],
```

Comment:
Hmm actually maybe our units are able to move sidewards, but only able to shoot straight or something similar.